main/lsp/json_types.cc

```diff
@@ -188,7 +188,7 @@ string TextDocumentContentChangeEvent::apply(string oldSource) const {
         // These offsets are non-nullopt assuming the input range is a valid range.
         auto startOffset = core::Loc::pos2Offset(old, start).value();
         auto endOffset = core::Loc::pos2Offset(old, end).value();
-        return oldSource.replace(startOffset, endOffset - startOffset, text);
+        return string(oldSource).replace(startOffset, endOffset - startOffset, text);
```

Comment:
Q: can this allocation be collapsed with the one on 187?

Reply:
No, unfortunately. Files require ownership over strings (&&):

https://github.com/sorbet/sorbet/blob/master/core/Files.h#L90

More info: https://www.tutorialspoint.com/What-is-double-address-operator-and-and-in-Cplusplus

I'd have to `move()` the string into the file, so I'd have to re-create it later.